cdatgui/editors/widgets/legend_widget.py

```diff
@@ -0,0 +1,552 @@
+from types import FunctionType
+
+from cdatgui.editors.widgets.dict_editor import DictEditorWidget
+from PySide import QtCore, QtGui
+from cdatgui.editors.model import legend
+from cdatgui.editors.preview.legend_preview import LegendPreviewWidget
+from cdatgui.bases.window_widget import BaseOkWindowWidget
+from cdatgui.editors.colormap import QColormapEditor
+from cdatgui.utils import pattern_thumbnail
+from cdatgui.bases.dynamic_grid_layout import DynamicGridLayout
+from functools import partial
+
+
+class CustomScrollArea(QtGui.QScrollArea):
```

Comment:
Add a comment and possibly a more descriptive name talking about why this class is necessary